src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LSPProjectSnapshotManagerDispatcher.cs

```diff
@@ -1,28 +1,11 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
-using System;
-using Microsoft.CodeAnalysis.Razor.Logging;
+using Microsoft.CodeAnalysis.Razor;
 using Microsoft.CodeAnalysis.Razor.Workspaces;
-using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer;
 
-internal class LSPProjectSnapshotManagerDispatcher : ProjectSnapshotManagerDispatcherBase
+internal class LSPProjectSnapshotManagerDispatcher(IErrorReporter errorReporter) : ProjectSnapshotManagerDispatcher(errorReporter)
 {
-    private const string ThreadName = "Razor." + nameof(LSPProjectSnapshotManagerDispatcher);
-
-    private readonly ILogger _logger;
-
-    public LSPProjectSnapshotManagerDispatcher(IRazorLoggerFactory loggerFactory) : base(ThreadName)
-    {
-        if (loggerFactory is null)
-        {
-            throw new ArgumentNullException(nameof(loggerFactory));
-        }
-
-        _logger = loggerFactory.CreateLogger<LSPProjectSnapshotManagerDispatcher>();
-    }
-
-    public override void LogException(Exception ex) => _logger.LogError(ex, ThreadName + " encountered an exception.");
```

Comment:
Note: We no longer create a custom `ILogger`. Instead, we rely on the language server's `IErrorReporter` implementation.